### PR TITLE
git-flow: initial draft commit

### PR DESCRIPTION
**Pull request by alex-pezzati**
- initial git flow guide draft with images
- setup for future expansion, if needed

### CODE REVIEW

File: git-flow-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+### What is the recommended git flow for short practice problems?
+
+-   In Module 2, morning practice sessions are short periods of time to tackle concepts through smaller problems. In the interest of time, you may find it simpler to work on your own code and repos for these shorter sessions while collaborating with your pair. Working together to understand and solve the problems but maximizing your own coding practice time as well.
+
+-   If you would prefer to commit to the same repos, you may do so. Attempt to coordinate with your pair in advance, prior to the designated practice time, to decide who creates the repos, send/accept links. This will minimize downtime.
+
+### What is the recommended git flow for long project problems?
+
+-   For longer practices it is recommended that pairs follow a/A suggested git flow of working from the same repo while taking turns swapping roles between driving and navigating.
```
Comment (chaunceykm) [inline git-flow-guide/README.md:163]: I would change recommended to expected. I don't want students to believe that pairing is an optional practice for afternoon practices.

**Review by chaunceykm — CHANGES_REQUESTED** [review]
minor correction needed

**Comment by barrymattern** [comment]
Alex, as I mentioned, this is coming along quite nicely – having this info all in one place/repo will be very valuable for the students. As I read through these were some things that came to mind. Let me know if you'd like to talk through any of them  👍

# Thoughts

- For step-by-step guides an idea I've been taught is to write your draft, then cut your text in half  ...and then cut it in half again
  - Goal is to get the steps at a quick glance
  - Then expand to a secondary or longer guide (or second page) to go more in-depth
- Students will more than likely be in a rush when reading this
  - Might be good to have 2 versions of this guide:
    1. Short/quick, step by step guide for students to follow in the moment
    2. Long/in-depth guide for TAs to update and review as needed
- There's some great info in here, and a nice layout
  - I'm wondering if these should be broken out into separate guides, then all connected in one main guide?
  - For example, initial README could have links to each quick secondary guide (like an index)

---

### Quick idea for student-facing guide to update remote URL:

1. Double check remote URL
```
git remote -v
``` 
2. If it's a/A's URL, update to your repo's URL
```
git remote set-url origin https://github.com/USERNAME/REPOSITORY.git
```
  - When done correctly it looks like nothing happens
3. Double check remote URL one more time
```
git remote -v
``` 
  - You may need to add and commit a change in order to push

These commands & more can be found in the [GitHub Docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)

File: git-flow-guide/README.md (hunk 1 above)
Comment (alex-pezzati) [reply git-flow-guide/README.md:163]: Made the requested changes. Thanks for the feedback!

**Comment by alex-pezzati** [comment]
> Alex, as I mentioned, this is coming along quite nicely – having this info all in one place/repo will be very valuable for the students. As I read through these were some things that came to mind. Let me know if you'd like to talk through any of them 👍
> 
> # Thoughts
> * For step-by-step guides an idea I've been taught is to write your draft, then cut your text in half  ...and then cut it in half again
>   
>   * Goal is to get the steps at a quick glance
>   * Then expand to a secondary or longer guide (or second page) to go more in-depth
> * Students will more than likely be in a rush when reading this
>   
>   * Might be good to have 2 versions of this guide:
>     
>     1. Short/quick, step by step guide for students to follow in the moment
>     2. Long/in-depth guide for TAs to update and review as needed
> * There's some great info in here, and a nice layout
>   
>   * I'm wondering if these should be broken out into separate guides, then all connected in one main guide?
>   * For example, initial README could have links to each quick secondary guide (like an index)
> 
> ### Quick idea for student-facing guide to update remote URL:
> 1. Double check remote URL
> 
> ```
> git remote -v
> ```
> 
> 1. If it's a/A's URL, update to your repo's URL
> 
> ```
> git remote set-url origin https://github.com/USERNAME/REPOSITORY.git
> ```
> 
> * When done correctly it looks like nothing happens
> 
> 1. Double check remote URL one more time
> 
> ```
> git remote -v
> ```
> 
> * You may need to add and commit a change in order to push
> 
> These commands & more can be found in the [GitHub Docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)

I was thinking I could circumvent the need for multiple guides with just efficient navigation links. I could add a "back to top of section" link as well. Just to move around between content effortlessly.

I do agree it is probably too wordy right now. Let me see what I can reduce it down to.

File: git-flow-guide/README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ TO DO:
 
 # Git Flow
 
-This is a guide with suggestions for _recommended_ practices for git and GitHub while in a/A's Module 2: Computer Science and continued studies at a/A. Let's git into it!
+This is a guide with for _expected_ practices for git and GitHub while in a/A's Module 2: Computer Science and continued studies at a/A. Let's git into it!
```
Comment (chaunceykm) [inline git-flow-guide/README.md:18]: I think this needs a look over. The phrasing seems a bit off.

**Review by chaunceykm — CHANGES_REQUESTED** [review]

**Review by chaunceykm — APPROVED** [review]

File: git-flow-guide/README.md (hunk 2 above)
Comment (alex-pezzati) [reply git-flow-guide/README.md]: Good catch. My fix before was too quick. Pushed. Moving onto reducing some of the bloat next.

**Comment by alex-pezzati** [comment]
Alright, I will make another pass on Monday although I already made a bunch of changes:

- reduced the word count in many places, 
- separated paragraphs/sentences into actionable bullet points, 
- added code blocks for easy copy/pasting, 
- added external links section, 
- added more comments as documentation for further expansion


This is seems like a pretty good state for this next cohort at least.
